notifier/tasks.py

```diff
@@ -104,6 +146,67 @@ def _time_slice(minutes, now=None):
     dt_start = dt_end - timedelta(minutes=minutes)
     return (dt_start, dt_end)
 
+
+@celery.task(max_retries=settings.DAILY_TASK_MAX_RETRIES, default_retry_delay=settings.DAILY_TASK_RETRY_DELAY)
+def do_forums_digests_flagged():
+    """
+    Generate and batch send digest emails for each thread with flagged posts.
+    """
+    def get_course_threads():
+        """
+        Parse threads from comments service API and group by course_id.
+
+        Returns:
+            dict: key=course_id, value=list of thread URLs with flagged posts
+        """
+        output = {}
+        for thread_url in get_flagged_threads():
+            match = re.search('^https?:\/\/\S+\/courses\/((?:[^\/]+\/){3})\S*', thread_url)
```

Comment:
See my comment on https://github.com/Stanford-Online/cs_comments_service/pull/1 about upgrading the API to return a more sophisticated payload. That would save us from using this brittle regex here.

Much like the `input_file` parameters, this regex matching is also a relic of the original Heroku/Rakefile worldview. At that time, we had to do regex matching because the only output we had access to was STDOUT from an already-existing Rakefile command. Now that we're writing a custom API endpoint, we should take advantage of this flexibility and tailor it to our needs.